### PR TITLE
feat: `@suggest` command and next step prompts

### PR DESCRIPTION
**Pull request by dustinbyrne**
The following changes support the ability to render "next steps" within the conversation view following a completed Navie response. The user can select a next step to automatically submit a generated prompt based on the most recent topic of conversation.

---

### `@packages/components`

- **Popper Component**:
  - Added hover, click, and mouse move functionality for better interaction.
  - Introduced a `timeToDisplay` prop to control the display time of elements when hovered.
  - Updated computed properties and methods to support the new behavior.

- **StreamingMessageContent Component**:
  - Integrated `VNextPromptButton` to handle user input prompts efficiently.

- **Chat Component**:
  - Routed `threadId` prop to provide context in messages.
  - Added methods `setInput` and `moveCursorToEnd` to manage user input more dynamically.

- **ChatInput Component**:
  - Added cursor control functionalities to enhance user experience.
  - Included method adjustments to better manage input fields.

- **NextPromptButton Component**:
  - New component created to handle user prompts for next steps.
  - Component supports commands and optional prompts, including `@click` and `@click.shift` events.

- **UserMessage Component**:
  - Added a section for next-step suggestions visible when content is complete.
  - Integrated `VNextPromptButton` and `VPopper` for interactive message prompts.
  - Watch for `complete` status to trigger `nextStepSuggestions`.
  
- **AppMapRPC Implementation**:
  - Added a new method `suggest(threadId: string)` to handle suggestion RPC calls.

---

### `@packages/navie`

- **Command Enhancements**:
  - Implemented `@suggest` command to generate JSON responses with next steps for user actions.
  
- **New `SuggestCommand`**:
  - Created a command to handle next-step classifications using the `NextStepClassificationService`.
  
- **Completion Service Updates**:
  - Updated `AnthropicCompletionService` and `OpenAICompletionService` to handle new structured output JSON schema.
  - Added methods to both services to support JSON schema-based responses.
  
- **NextStepClassificationService**: 
  - New service to handle the scoring and generation of next-step suggestions for user interactions.

- **ModelType Enum**:
  - Introduced `ModelType` enumeration to differentiate between `Full` and `Mini` models, enabling better control over model selection based on performance and intelligence requirements.
  - Completions generated via official OpenAI API endpoints will default to `gpt-4o-mini`. Otherwise, the mini model will fall back to the base model (`APPMAP_NAVIE_MODEL`).
  - The mini model can be customized by setting `APPMAP_NAVIE_MINI_MODEL`. This is optional.

---

### `@packages/cli`

- **AppMapRPC Updates**:
  - Added `suggest` method to the AppMapRPC handlers, integrating the new suggestion functionality.
  
- **Navie SuggestHandler**:
  - Created an RPC handler for `v1.navie.suggest`

---

### General Updates

- **Dependency Updates**:
  - Updated multiple dependencies, including `openai`, `langchain`, and `@anthropic-ai/sdk` for improved functionality.
  - Added `zod` dependency to handle new JSON schema validations.

### CODE REVIEW

File: packages/rpc/src/navie.ts [hunk 1]
```diff
@@ -16,5 +16,19 @@ export namespace NavieRpc {
...
+    export namespace Suggest {
+      export const Method = 'v1.navie.suggest';
+      export type NextStep = {
+        command: 'generate' | 'diagram' | 'plan' | 'test' | 'explain' | 'help';
+        prompt: string;
+        label: string;
+        overallScore: number;
+      };
+      export type Params = {
+        threadId: string;
```
Comment (kgilpin) [inline packages/rpc/src/navie.ts:29]: Would this be on the most recent assistant id or user message id, to get a suggestion from a specific point in the conversation?

File: packages/cli/src/rpc/navie/suggest.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  threadId: string
+): Promise<NavieRpc.V1.Suggest.Response> {
+  const navie = navieProvider(NOP, NOP, NOP);
+  return new Promise((resolve, reject) => {
+    navie
+      .on('token', (response: string) =>
+        resolve(JSON.parse(response) as NavieRpc.V1.Suggest.Response)
+      )
+      .on('error', reject)
+      .ask(threadId, '@suggest /nocontext /nohelp', undefined, undefined)
```
Comment (kgilpin) [inline packages/cli/src/rpc/navie/suggest.ts:19]: I am pretty sure that only `@explain` mode can auto-trigger `@help` but I guess `/nohelp` can't hurt.

File: packages/navie/src/services/next-step-classification-service.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+const SYSTEM_PROMPT = `Your job is to predict the next step(s) for the user given a transcript of their conversation. The transcript will contain a single question and answer, already presented to the user. The users next action will typically be in response to the assistant's answer.
+
+## Commands
+These are the actions that the user may request next:
+- \`generate\`: This will generate code for the user.
+- \`diagram\`: This will generate a (sequence, flow, class, entity relation) diagram for the user. This is useful for visualizing complex concepts, relationships, or processes.
+- \`plan\`: This is useful to create step by step plans to implement a code change in the user's application.
+- \`test\`: This action will generate test cases for the user.
+- \`explain\`: This action will explain parts of the codebase or application behavior.
+- \`help\`: This action will teach the user how to use AppMap Navie.
```
Comment (kgilpin) [inline packages/navie/src/services/next-step-classification-service.ts:38]: A couple of suggestions on this:

1) Put the commands in the general order that the user will use them; for example, plan before generate.
2) In the prompt, propose a few things to the LLM that are fairly concrete. For example, "@generate will often follow @plan". "@explain will often be followed by @diagram". "@test will often follow @generate". Etc.

**Review by kgilpin — APPROVED** [review]
Thanks, this is really nice. I'm not that proficient with UI changes - so maybe @dividedmind can review for UI, or we can merge as-is?

File: packages/rpc/src/navie.ts (hunk 1 above)
Comment (dustinbyrne) [reply packages/rpc/src/navie.ts:29]: Currently this API reflects the capabilities of the UI, where next steps are only displayed for the most recent assistant response. It won't be difficult to transition to a state where we can request next steps from any point in the future if need be.

**Pull request merged**